Bot/assistant_bot.py

```diff
@@ -1,57 +1,122 @@
+import asyncio
+from datetime import datetime
 import discord
+from discord.ext import tasks
 import os
 import requests
 import json
 import random
+from dotenv import load_dotenv
 
+# load .env file from local
+load_dotenv()
 # instance of the client
 Intents = discord.Intents.default()
-# Intents.message_content = True
+Intents.message_content = True
 client = discord.Client(intents=Intents)
 
-sad_words = ["sad", "depressed", "unhappy", "angry", "miserable", "depressing"]
+meme_channel_id = int(os.getenv('MEME_CHANNEL_ID'))
 
+# const
+sad_words = ["sad", "depressed", "unhappy", "angry", "miserable", "depressing"]
 starter_encouragements = [
-    "Cheer up!", 
+    "Cheer up!",
     "You can do it!",
     "Hang in there!",
     "You are a star!",
     "You are a great person!"
 ]
+zen_url = "https://zenquotes.io/api/random"
+meme_url = "https://meme-api.herokuapp.com/gimme/1"
+
+# method part
+
 
 def get_quote():
     """
     This function gets a random quote from the API
+    return value is {'q': ~~~, 'a': ~~~, 'h': ~~~}
     """
-    url = "https://zenquotes.io/api/random"
-    response = requests.get(url)
+    headers = {'user-agent': 'vscode-restclient'}
+    response = requests.request("GET", zen_url, headers=headers)
     json_data = json.loads(response.text)
-    quote = json_data[0]['q'] + " - " + json_data[0]['a']
-    return quote
+    if not json_data[0]:
+        return "failed fetching data... Anyway, you can do it!!!"
+    return json_data[0]
+
+
+def get_meme():
+    """
+    This function gets a random meme from the API
+    get a jpg or png url
+    please refer to this below url
+    https://github.com/D3vd/Meme_Api
+    """
+    response = requests.request("GET", meme_url)
+    json_data_for_meme = json.loads(response.text)
+    url = json_data_for_meme['memes'][0]['url']
+    return url
+
+
+@tasks.loop(minutes=1.0)
+async def show_meme_daily():
+    """
+    This function sends a meme at 8:00 am everyday.
+    This task is called every 1 minutes and if the time is 8:00, it will send meme.
+    """
+    now = datetime.now().strftime('%H:%M')
```

Comment:
Oh oh, timezones.

This may work on your machine where time is in our current timezone, but if this is hosted somewhere we can't guarentee that. 

We might want to specify the timezone like this
```suggestion
    now = datetime.now(tz=zoneinfo.ZoneInfo("Australia/Brisbane")).strftime('%H:%M')
```
and import `zoneinfo` at the top